webapp/src/containers/modals/SubscribeModal/filters/repos.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {useEffect, useMemo} from 'react';
+
+import pluginConstants from 'pluginConstants';
+import {filterLabelValuePairAll} from 'pluginConstants/common';
+
+import Dropdown from 'components/dropdown';
+
+import usePluginApi from 'hooks/usePluginApi';
+
+import {formLabelValuePairs} from 'utils';
+
+type ReposFilterProps = {
+    handleSelectRepo: (repo: string) => void
+    selectedRepo: string
+} & ReposSubscriptionFiltersRequest
+
+const ReposFilter = ({handleSelectRepo, project, organization, selectedRepo}: ReposFilterProps) => {
+    const {
+        getApiState,
+        makeApiRequestWithCompletionStatus,
+    } = usePluginApi();
+
+    const reposSubscriptionFiltersRequest = useMemo<ReposSubscriptionFiltersRequest>(() => ({
+        organization,
+        project,
+    }), [organization, project]);
+
+    useEffect(() => {
+        makeApiRequestWithCompletionStatus(
+            pluginConstants.pluginApiServiceConfigs.getRepositories.apiServiceName,
+            reposSubscriptionFiltersRequest,
+        );
+    }, [reposSubscriptionFiltersRequest]);
+
+    const {data, isLoading: isGetRepositoriesLoading, isError: isGetRepositoriesError} = getApiState(
+        pluginConstants.pluginApiServiceConfigs.getRepositories.apiServiceName,
+        reposSubscriptionFiltersRequest as APIRequestPayload,
+    );
+    const repositoriesData = data as ReposSubscriptionFiltersResponse[] || [];
```

Comment:
can data be `null` | `undefined`?

Reply:
yes it can be, and the reason we didn't use `??` here is to cover all possible falsy values instead of just `null, undefined`